### PR TITLE
Switch from lerna to npm workspaces

### PR DESCRIPTION
**Pull request by bingenito**
Remove direct dependency to lerna and use native npm workspaces for dependency management.

### CODE REVIEW

File: .gulp/tasks/tests.js [hunk 1]
```diff
@@ -16,9 +16,6 @@ module.exports = function (gulp, config) {
                 // Run jasmine under istanbul
                 gulp.src(config.test.src)
                     .pipe(jasmine({ verbose: true, errorOnFail: true, includeStackTrace: false }))
-                    .on('error', (err) => {
```
Comment (psmulovics) [inline .gulp/tasks/tests.js:19]: Is this now a built in functionality?

File: package.json [hunk 2]
```diff
@@ -5,22 +5,21 @@
...
-    "build": "lerna run build:lerna --stream",
+    "clean": "npm run clean --workspaces --if-present",
+    "build": "npm run build --workspaces --if-present",
     "test": "gulp test",
     "start": "gulp server",
     "docs": "typedoc",
-    "watch": "lerna run --parallel watch",
-    "deploy": "lerna run deploy"
+    "deploy": "npm run deploy --workspaces --if-present",
+    "version": "npx lerna version --no-push --no-git-tag-version"
```
Comment (psmulovics) [inline package.json:16]: If we can replace this with something better, we could remove lerna.json and the whole dependency on it, isn't it?

File: .gulp/tasks/tests.js (hunk 1 above)
Comment (bingenito) [reply .gulp/tasks/tests.js:19]: Look at line above line, i'm hoping it was fixed at some point as it was a regression causing it to ignore errors at one point. If we have tests fail and it doesn't error the PR then we find out.  Gulp is next on chopping block anyway.

File: package.json (hunk 2 above)
Comment (bingenito) [reply package.json:16]: Absolutely 100%.  I had lerna config deleted originally but then after a few minutes searching I found nothing better to help with versioning so added this npm script in to help facilitate that.  I really like how it prompts if you want patch, minor, major.

**Pull request merged**